arm64: dts: qcom: msm8916-ufi: make UDC dual mode.

185264646 opened the pull request:
There are some requests for switching the UDC to host mode, but dr_mode="peripheral" prevents to do so. Remove dr_mode="peripheral" so that it defaults to otg mode and can be switched to host mode in userspace.

Not tested yet, though.

185264646 committed:
arm64: dts: qcom: msm8916-ufi: make UDC dual mode

It's possible to use this device with a (non-standard) hub to get USB
working in host mode, but dr_mode="peripheral" prevents the UDC to
do so.
Remove dr_mode="peripheral"  and add usb-role-switch so that it defaults
to otg mode and can be switched to host mode in userspace.

Signed-off-by: Yang Xiwen <forbidden405@outlook.com>

TravMurav commented:
Very tiny nitpick but in the commit message I'd motivate the change like

"It's possible to use this device with ... to get usb ... but forced peripheral mode prevents ... switch in userspace ..."

instead of motivating it with "someone wants"

Otherwise I think this should be a fine change after you can take it for a spin to see if it doesn't immediately break down. (I don't remember what was it about on switchig the usb mode from the userspace and if we need any extra extcon magic here to make sure it doesn't explode)

185264646 commented:
> Very tiny nitpick but in the commit message I'd motivate the change like
> 
> "It's possible to use this device with ... to get usb ... but forced peripheral mode prevents ... switch in userspace ..."
> 
> instead of motivating it with "someone wants"
> 
> Otherwise I think this should be a fine change after you can take it for a spin to see if it doesn't immediately break down. (I don't remember what was it about on switchig the usb mode from the userspace and if we need any extra extcon magic here to make sure it doesn't explode)

Yes, so I mark it WIP and a draft. There are already a lot of hubs designed for that and they are confirmed to work. However, the commonly used dts(in OpenStick project), used a fake ID-pin for that rather than pm8916_usbin, so I don't know if it also works with our device tree. I would wait for someone to take a test.

185264646 commented:
It is reported that this patch works as intended.

185264646 committed:
# Lines starting with # will be removed from the cover letter. You can
# use them to add notes or reminders to yourself.

# You can add trailers to the cover letter. Any email addresses found in
# these trailers will be added to the addresses specified/generated
# during the b4 send stage. You can also run "b4 prep --auto-to-cc" to
# auto-populate the To: and Cc: trailers based on the code being
# modified.

To: Andy Gross <agross@kernel.org>
To: Bjorn Andersson <andersson@kernel.org>
To: Konrad Dybcio <konrad.dybcio@linaro.org>
To: Rob Herring <robh+dt@kernel.org>
To: Krzysztof Kozlowski <krzysztof.kozlowski+dt@linaro.org>
Cc: linux-arm-msm@vger.kernel.org
Cc: devicetree@vger.kernel.org
Cc: linux-kernel@vger.kernel.org
Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>

--- b4-submit-tracking ---
# This section is used internally by b4 prep for tracking purposes.
{
  "series": {
    "revision": 1,
    "change-id": "20230506-msm8916-fix_usb-4947eb5b2820",
    "base-branch": "msm8916/6.3",
    "prefixes": []
  }
}

stephan-gh commented:
Seems like a forced line break sneaked in for the patch you sent upstream: https://lore.kernel.org/linux-arm-msm/TYZPR04MB632102315884225B7343533B96729@TYZPR04MB6321.apcprd04.prod.outlook.com/

```
diff --git a/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
b/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
```

this should be on one line otherwise it doesn't apply correctly. Please fix this and resend the patch.

If you cannot send the patch directly with the SMTP server through b4, I would recommend using the web submission endpoint of b4 as described in https://b4.docs.kernel.org/en/latest/contributor/send.html. By sending the patches directly through b4 you ensure that they won't be mangled by other email clients.

185264646 commented:
> Seems like a forced line break sneaked in for the patch you sent upstream: https://lore.kernel.org/linux-arm-msm/TYZPR04MB632102315884225B7343533B96729@TYZPR04MB6321.apcprd04.prod.outlook.com/
> 
> ```
> diff --git a/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
> b/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
> ```
> 
> this should be on one line otherwise it doesn't apply correctly. Please fix this and resend the patch.
> 
> If you cannot send the patch directly with the SMTP server through b4, I would recommend using the web submission endpoint of b4 as described in https://b4.docs.kernel.org/en/latest/contributor/send.html. By sending the patches directly through b4 you ensure that they won't be mangled by other email clients.

I'm too lazy to setup b4 on another PC. Will resend sometime later.

185264646 commented:
> Seems like a forced line break sneaked in for the patch you sent upstream: https://lore.kernel.org/linux-arm-msm/TYZPR04MB632102315884225B7343533B96729@TYZPR04MB6321.apcprd04.prod.outlook.com/
> 
> ```
> diff --git a/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
> b/arch/arm64/boot/dts/qcom/msm8916-ufi.dtsi
> ```
> 
> this should be on one line otherwise it doesn't apply correctly. Please fix this and resend the patch.
> 
> If you cannot send the patch directly with the SMTP server through b4, I would recommend using the web submission endpoint of b4 as described in https://b4.docs.kernel.org/en/latest/contributor/send.html. By sending the patches directly through b4 you ensure that they won't be mangled by other email clients.

It is accepted by upstream now, maybe close this as completed.

stephan-gh commented:
I can backport this in the next branch so you can make use of it more easily before it arrives in a release upstream :)

stephan-gh commented:
Backported the upstream commit in `msm8916/6.4-rc4`, thanks!
